js/travelblog.js

```diff
@@ -111,67 +111,50 @@ function createEntry(entry, index){
         `;
 
       console.log(blogEntry);
-      button.innerHTML = "🗑️";
-      button.classList.add("delete-button");
-      blogEntry.appendChild(button);
-      button.addEventListener("click", function() {
+      deleteButton.innerHTML = "🗑️";
+      deleteButton.classList.add("delete-button");
+      blogEntry.appendChild(deleteButton);
+      deleteButton.addEventListener("click", function() {
         deleteEntry(index);
         putInHTML();
       });
 
-      button2.innerHTML = `<span class="edit-icon">${editEmoji} </span>`;
-      button2.classList.add("edit-button");
-      blogEntry.appendChild(button2);
-
-      button2.addEventListener("click", function(){
-          if(editableCity[index].contentEditable === "false"){
-             editableCity[index].contentEditable = 'true';
-          } else{
-            editableCity[index].contentEditable = "false";
-          };
-
-          if(editableCountry[index].contentEditable === "false"){
-              editableCountry[index].contentEditable = "true";
-          }else{
-            editableCountry[index].contentEditable = "false";
-          };
-
-          if(editableTime[index].contentEditable === "false"){
-              editableTime[index].contentEditable = "true";
-          }else{
-            editableTime[index].contentEditable = "false";
-          };
-
-
-          if(editablePurpose[index].contentEditable === "false"){
-              editablePurpose[index].contentEditable = "true";
-          }else{
-            editablePurpose[index].contentEditable = "false";
-          };
-
-          if(imageLinks[index].style.display == "none"){
+      editButton.innerHTML = `<span class="edit-icon">${editEmoji} </span>`;
+      editButton.classList.add("edit-button");
+      blogEntry.appendChild(editButton);
+
+      editButton.addEventListener("click", function(){
+          // Boolean("false") === true; Boolean("true") === false;
+          // true.toString() === "true"; false.toString() === "false"
+          const negateBooleanString = (booleanString) => booleanString === "true" ? "false" : "true"
```

Comment:
anstatt sich wiederholende if-else-Blöcke zu bauen, einfach eine Funktion die den gewünschten Wert zurückgibt.

Würde ich im echten Leben generell anders lösen, in dem Kontext aber ist das schon angebracht.

Reply:
Wobei ich generell nicht verstehen, warum du die Sachen nicht einfach vom Typ Boolean hast, anstatt vom Typ String. Für JSON.stringify und JSON.parse macht das kein Unterschied.